views/components/Ad.jsx

```diff
@@ -51,14 +53,52 @@ class Ad extends React.Component {
   }
 
   componentDidMount () {
-    var ctx = this;
-
     this.getAd().then((ad) => {
+      this.adObject = new models.Link(ad).toJSON();
       return this.setState({
         loaded: true,
-        ad: new models.Link(ad).toJSON(),
+        ad: this.adObject,
       });
     }, () => {});
+
+    window.addEventListener('scroll', this._onScroll);
```

Comment:
This and resize should be throttled so it isn't calling so many times; try out lodash's throttle, maybe at 100ms.